src/planner/expression_binder/check_binder.cpp

```diff
@@ -41,10 +42,28 @@ BindResult ExpressionBinder::BindQualifiedColumnName(ColumnRefExpression &colref
 	return BindExpression(&result, 0);
 }
 
-BindResult CheckBinder::BindCheckColumn(ColumnRefExpression &colref) {
+// FIXME: this is quite naive and optimistic
+bool IsLambdaBinding(vector<DummyBinding> *lambda_bindings_p, ColumnRefExpression &colref) {
+	if (!lambda_bindings_p) {
+		return false;
+	}
+	auto &lambda_bindings = *lambda_bindings_p;
+	for (auto &binding : lambda_bindings) {
+		if (binding.name_map.count(colref.GetColumnName())) {
```

Comment:
Nit: we only need one key, no? So `.contains` instead of `.count`?

Reply:
Ah fair, that semantically makes more sense, they do the same thing though, it's either 1 or 0

Reply:
Yes, but they should have different complexities, no? I am just guessing but `contains` probably does a O(1) lookup in the hash table, whereas `count` is O(n)? but only in a multimap, ah well, I don't know haha

Reply:
`contains` is sadly not a thing from what I can find
it's an unordered_set so it should be a hash map internally, and this should be O(1)
https://cplusplus.com/reference/unordered_set/unordered_set/count/ says average cast is constant 👍